core/src/memory.rs

```diff
@@ -543,6 +543,10 @@ pub enum AllocError {
     /// outside additionally allocated for this program.
     #[display(fmt = "Page {_0} cannot be freed by the current program")]
     InvalidFree(u32),
+    /// Gas charge error
+    #[from]
+    #[display(fmt = "{_0}")]
+    GasChargeError(ChargeError),
```

Comment:
```suggestion
    Charge(ChargeError),
```
Consistency with other error variants